pallets/gear-messenger/src/migrations.rs

```diff
@@ -0,0 +1,435 @@
+                initialized,
+                awaken,
+                reply_sent: random(),
+                reservation_nonce: Default::default(),
+                system_reservation: if random() { Some(random()) } else { None },
+            })
+        };
+        v2::StoredDispatch {
+            kind,
+            message: StoredMessage::new(
+                MessageId::from(random::<u64>()),
+                ProgramId::from(random::<u64>()),
+                ProgramId::from(random::<u64>()),
+                random_payload(),
+                random(),
+                details,
+            ),
+            context,
+        }
+    }
+
+    #[test]
+    fn migration_to_v3_works() {
+        new_test_ext().execute_with(|| {
+            StorageVersion::new(2).put::<GearMessenger>();
+
+            let waitlist = up_to(32, || {
+                (
+                    ProgramId::from(random::<u64>()),
+                    MessageId::from(random::<u64>()),
+                    random_dispatch(false),
+                    Interval {
+                        start: random(),
+                        finish: random(),
+                    },
+                )
+            })
+            .collect::<Vec<_>>();
+
+            for dispatch in waitlist.iter() {
+                v2::Waitlist::<Test>::insert(
+                    dispatch.0,
+                    dispatch.1,
+                    (dispatch.2.clone(), dispatch.3.clone()),
+                );
+            }
+
+            let dispatches = up_to(32, || {
+                (
+                    MessageId::from(random::<u64>()),
+                    random_dispatch(false),
+                    if random() {
+                        Some(MessageId::from(random::<u64>()))
+                    } else {
+                        None
+                    },
+                )
+            })
+            .collect::<Vec<_>>();
+
+            for dispatch in dispatches.iter() {
+                v2::Dispatches::<Test>::insert(
+                    dispatch.0,
+                    LinkedNode {
+                        next: dispatch.2,
+                        value: dispatch.1.clone(),
+                    },
+                );
+            }
+
+            let dispatch_stash = up_to(32, || {
+                (
+                    MessageId::from(random::<u64>()),
+                    random_dispatch(true),
+                    Interval {
+                        start: random(),
+                        finish: random(),
+                    },
+                )
+            })
+            .collect::<Vec<_>>();
+
+            for dispatch in dispatch_stash.iter() {
+                v2::DispatchStash::<Test>::insert(
+                    dispatch.0,
+                    (dispatch.1.clone(), dispatch.2.clone()),
+                );
+            }
+
+            let state = MigrateToV3::<Test>::pre_upgrade().unwrap();
+            let _ = MigrateToV3::<Test>::on_runtime_upgrade();
```

Comment:
Assert weight is not zero

Reply:
I did, a while ago, but it fails, because, Test weights for everything is zero, so it always adds zero for every read and write.

Reply:
https://github.com/gear-tech/gear/blob/ed3290bd7440103deb39dbfc715c3bc6794e90be/pallets/gear-messenger/src/mock.rs#L50

Replace to`DbWeight = RocksDbWeight`